google/cloud/forseti/scanner/audit/bigquery_rules_engine.py

```diff
@@ -243,12 +266,33 @@ def __init__(self, rule_name, rule_index, rules):
         Args:
             rule_name (str): Name of the loaded rule.
             rule_index (int): The index of the rule from the rule definitions.
-            rules (dict): The rules from the file.
+            rules (RuleReference): The rules from the file and corresponding
+                values.
         """
         self.rule_name = rule_name
         self.rule_index = rule_index
         self.rules = rules
 
+    def _is_applicable(self, bigquery_acl):
+        """Determine whether the rules are applicable to the given acl.
+
+         Args:
+            bigquery_acl (BigqueryAccessControls): BigQuery ACL resource.
+         Returns:
+            bool: True if the rules are applicable to the given acl, False
+                otherwise.
+        """
+        rule_bigquery_acl = self.rules.bigquery_acl
+        rule_regex_to_val = {
+            rule_bigquery_acl.dataset_id: bigquery_acl.dataset_id,
+            rule_bigquery_acl.role: bigquery_acl.role,
+        }
+
+        return all([
+            re.match(rule_regex, acl_val)
+            for (rule_regex, acl_val) in rule_regex_to_val.iteritems()
+        ])
```

Comment:
It was already an issue, but the list comprehension will undo the laziness of all, so this will always evaluate the second re.match. 

For here and all_matched below, it might be better to use a list of tuples instead of a dict, and a helper function
```
  def all_re_match(patterns_and_strings):
    for pattern, string in patterns_and_strings):
      if not re.match(pattern, string):
        return False
    return True
```
...
```
  return all_re_match([
      (rule_bigquery_acl.dataset_id, bigquery_acl.dataset_id),
      (rule_bigquery_acl.role, bigquery_acl.role)
  ])
```

It may even make sense to put the helper function in common/util/regular_exp.py

Reply:
Done.

I was not sure if the list of tuples was required, so I just kept it as a dict as it looks cleaner.